pkg/controllers/drift/controller.go

```diff
@@ -0,0 +1,133 @@
+package drift
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/karpenter-core/pkg/apis/provisioning/v1alpha5"
+	operatorcontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/cloudprovider"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/logging"
+	"net/http"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"time"
+)
+
+const controllerName = "drift"
+
+type Drift struct {
+	kubeClient    client.Client
+	cloudProvider *cloudprovider.CloudProvider
+}
+
+// NewController constructs a controller instance
+func NewController(kubeClient client.Client, cloudProvider *cloudprovider.CloudProvider) *Drift {
+	return &Drift{
+		kubeClient:    kubeClient,
+		cloudProvider: cloudProvider,
+	}
+}
+
+func (d Drift) Reconcile(ctx context.Context, req controllerruntime.Request) (reconcile.Result, error) {
+	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named(controllerName).With("provisioner", req.Name))
+	node := &v1.Node{}
+	if err := d.kubeClient.Get(ctx, req.NamespacedName, node); err != nil {
+		return reconcile.Result{}, fmt.Errorf("getting node, %w", err)
+	}
+
+	provisionerName, provisionerExists := node.Labels[v1alpha5.ProvisionerNameLabelKey]
+	if !provisionerExists {
+		return reconcile.Result{}, nil
+	}
+
+	if drifted, ok := node.Annotations[v1alpha5.DriftedAnnotationKey]; ok && drifted == "true" {
+		return reconcile.Result{}, nil
+	}
+
+	provisioner := &v1alpha5.Provisioner{}
+	if err := d.kubeClient.Get(ctx, types.NamespacedName{Name: provisionerName}, provisioner); err != nil {
+		return reconcile.Result{}, fmt.Errorf("getting provisioner, %w", err)
+	}
+
+	if drifted, err := d.cloudProvider.IsNodeDrifted(ctx, provisioner, *node); err != nil {
+		return reconcile.Result{}, fmt.Errorf("getting drift for node, %w", err)
+	} else if drifted {
+		stored := node.DeepCopy()
+		node.Annotations[v1alpha5.DriftedAnnotationKey] = "true"
+		if err := d.kubeClient.Patch(ctx, node, client.MergeFrom(stored)); err != nil {
+			return reconcile.Result{RequeueAfter: 60 * time.Second}, fmt.Errorf("patching node, %w", err)
```

Comment:
i think since we're returning an error, we shouldn't need to put in a requeueAfter.

Reply:
Done.